src/Serilog/Policies/DictionaryDestructuringPolicy.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright 2013-2021 Serilog Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using Serilog.Core;
+using Serilog.Events;
+
+namespace Serilog.Policies
+{
+    class DictionaryDestructuringPolicy<TKey, TValue> : IDestructuringPolicy
+    {
+        public bool TryDestructure(object value, ILogEventPropertyValueFactory propertyValueFactory, out LogEventPropertyValue result)
+        {
+            if (value is IDictionary dict)
```

Comment:
Not all dictionaries will support this; the generic case should at least use `IDictionary<K,V>`

Reply:
This check works for both `IDictionary` and `IDictionary<K,V>`.

Reply:
This works for `Dictionary<K,V>` which implements `IDictionary`, but other types that implement `IDictionary<K,V>` don't all implement the non-generic version (see https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2?view=net-5.0).

Reply:
I understand now. Thanks 🙂